### ISSUE TITLE
`Gini_Spatial`and `TheilD` disagree on whether within/between inequality is larger?

### ISSUE THREAD

**Issue by LSYS** [pysal/inequality#68]
Hi there, thanks so much for making the computing of inequality indices easy. I'm using the package (example code below) to compare within vs between groups inequality and noticed that the Spatial Gini and the Theil decomposition measures tend to disagree on whether the within or between group inequality is larger.

Here is the preamble to prep the data:
```python
import libpysal
import numpy as np
from inequality.theil import TheilD
from inequality.gini import Gini_Spatial
import pandas as pd

f = libpysal.io.open(libpysal.examples.get_path('mexico.csv'))
periods = ["".join(["yr", str(ix)]) for ix in range(1,13)]

df = (
    pd.DataFrame(f, columns=["area", *periods])
    .assign(regime=np.array(f.by_col('hanson98')).astype(int))
)
df
```

|    | area                |   yr1 |   yr2 |   yr3 |   yr4 |   yr5 |   yr6 |   yr7 |   yr8 |   yr9 |   yr10 |   yr11 |   yr12 |   regime |
|---:|:--------------------|------:|------:|------:|------:|------:|------:|------:|------:|------:|-------:|-------:|-------:|---------:|
|  0 | Aguascalientes      | 10384 |  6234 |  8714 | 16078 | 21022 | 20787 | 27782 |     2 |     2 |      3 |      4 |      4 |        2 |
|  1 | Baja California     | 22361 | 20977 | 17865 | 25321 | 29283 | 26839 | 29855 |     1 |     1 |      5 |      1 |      1 |        1 |
|  2 | Baja California Sur |  9573 | 16013 | 16707 | 24384 | 29038 | 25842 | 26103 |     2 |     2 |      6 |      1 |      1 |        2 |
|  3 | Campeche            |  3758 |  4929 |  5925 | 10274 | 12166 | 51123 | 36163 |     6 |     5 |      4 |      5 |      5 |        5 |
|  4 | Chiapas             |  2934 |  4138 |  5280 |  7015 | 16200 |  8637 |  8684 |     5 |     5 |      7 |      5 |      5 |        5 |
|  5 | Chihuahua           |  8578 | 13997 | 16265 | 19178 | 23399 | 25332 | 30735 |     1 |     1 |      5 |      1 |      2 |        1 |
|  6 | Coahuila            |  8537 |  9673 | 12318 | 20562 | 25688 | 26084 | 28460 |     1 |     1 |      5 |      2 |      2 |        1 |
|  7 | Colima              |  6909 |  6049 |  6036 | 12551 | 17427 | 18313 | 21358 |     3 |     3 |      6 |      4 |      4 |        3 |
|  8 | Distrito Federal    | 17816 | 17119 | 23174 | 32386 | 42028 | 43810 | 54349 |     4 |     4 |      1 |      3 |      3 |        4 |
|  9 | Durango             | 12132 |  8859 |  9323 | 12700 | 16726 | 17353 | 17379 |     2 |     2 |      3 |      1 |      2 |        2 |
| 10 | Guanajuato          |  4359 |  5686 |  8209 | 11635 | 13864 | 13607 | 15585 |     3 |     3 |      3 |      4 |      4 |        3 |
| 11 | Guerrero            |  2181 |  3629 |  4991 |  6497 |  8727 |  9084 | 11820 |     5 |     5 |      7 |      5 |      5 |        5 |
| 12 | Hidalgo             |  4414 |  5194 |  6399 |  7767 | 12391 | 13091 | 12348 |     3 |     3 |      2 |      3 |      3 |        3 |
| 13 | Jalisco             |  5309 |  8232 |  9953 | 16288 | 20659 | 20133 | 21610 |     3 |     3 |      6 |      4 |      4 |        3 |
| 14 | Mexico              |  3408 |  4972 |  9053 | 17164 | 20165 | 18547 | 16322 |     4 |     4 |      1 |      3 |      3 |        4 |
| 15 | Michoacan           |  3327 |  5272 |  5244 |  8109 | 11206 | 10980 | 11838 |     3 |     3 |      7 |      4 |      4 |        3 |
| 16 | Morelos             |  6936 |  8962 | 10499 | 13892 | 16513 | 17701 | 18170 |     3 |     3 |      2 |      3 |      3 |        3 |
| 17 | Nayarit             |  4836 |  7515 |  7621 | 10880 | 13354 | 12757 | 11478 |     2 |     2 |      6 |      4 |      4 |        2 |
| 18 | Nuevo Leon          |  9073 | 11490 | 20117 | 28206 | 34856 | 34726 | 38672 |     1 |     1 |      5 |      2 |      2 |        1 |
| 19 | Oaxaca              |  1892 |  4538 |  4140 |  5230 |  7730 |  8465 |  9010 |     5 |     5 |      7 |      5 |      5 |        5 |
| 20 | Puebla              |  3569 |  6415 |  6542 |  9775 | 13374 | 11895 | 15685 |     3 |     3 |      2 |      3 |      5 |        3 |
| 21 | Quertaro            | 11016 |  5560 |  7110 | 14073 | 20088 | 22441 | 26149 |     3 |     3 |      3 |      3 |      4 |        3 |
| 22 | Quintana Roo        | 21965 | 28747 |  9677 | 17046 | 26695 | 25049 | 33442 |     6 |     5 |      4 |      5 |      5 |        5 |
| 23 | San Luis Potosi     |  4372 |  7533 |  6440 |  9721 | 12691 | 15436 | 15866 |     2 |     2 |      3 |      4 |      4 |        2 |
| 24 | Sinaloa             |  4840 |  6663 |  9613 | 14477 | 15312 | 15823 | 15242 |     2 |     2 |      6 |      1 |      1 |        2 |
| 25 | Sonora              |  6399 | 10345 | 12134 | 22662 | 23181 | 24784 | 24068 |     1 |     1 |      5 |      1 |      1 |        1 |
| 26 | Tabasco             |  2459 |  3857 |  6494 |  9367 | 42361 | 16055 | 13360 |     6 |     5 |      4 |      5 |      5 |        5 |
| 27 | Tamaulipas          |  7508 |  8536 |  8383 | 17128 | 21937 | 19983 | 23546 |     1 |     1 |      5 |      2 |      2 |        1 |
| 28 | Tlaxcala            |  3605 |  4178 |  4357 |  6245 |  9882 | 10339 | 11701 |     3 |     3 |      2 |      3 |      3 |        3 |
| 29 | Veracruz            |  5203 | 10143 | 11404 | 12240 | 14252 | 13796 | 12191 |     3 |     3 |      4 |      5 |      5 |        3 |
| 30 | Yucatan             |  7990 |  8428 | 10067 | 11665 | 15239 | 13979 | 17509 |     6 |     5 |      4 |      5 |      5 |        5 |
| 31 | Zacatecas           |  3734 |  6435 |  5821 |  7426 |  8876 | 11656 | 11130 |     2 |     2 |      3 |      4 |      4 |        2 |

Decomposing within vs between inequality:

With Theil:
```python
theil = TheilD(df["yr1"], df["regime"])
theil.wg.item(), theil.bg.item()
```
(0.17435454225688188, 0.03458890004670406)

With Spatial Gini:

```python
w = libpysal.weights.block_weights(df["regime"], silence_warnings=True)
np.random.seed(0)
gs = Gini_Spatial(df["yr1"], w)

(gs.g - gs.wcg_share), gs.wcg_share
```
(0.05970572377548328, 0.2940179879590452)

Based on the above, Theil suggests the within-group inequality is larger, but the Spatial Gini suggests the between-group inequality is larger. Is this interpretation correct? Sorry if I missed something obvious that will help clear my confusion.

**Comment by sjsrey** [comment]
The `Gini_Spatial` and `TheilD` indices measure different types of decompositions and are based on distinct mathematical formulations. The Theil index is an entropy-based measure, while the Gini index is based on the relative mean absolute difference between all pairs of economies. In the case of the Gini index, the decomposition categorizes pairs  of economies into two groups: those that are "neighbors" and those pairs that are not neighbors. Conversely, the `TheilD` decomposition categorizes the economies (not the pairs) into different groups based on regional definitions. Due to the distinct underlying logic of these two decompositions, their results are not directly comparable, as they measure different aspects of spatial inequality.

For further information see:

- Rey, S. J., & Smith, R. J. (2013). "A spatial decomposition of the Gini coefficient." Letters in Spatial and Resource Sciences, 6(2), 55-70. 
- Rey, Sergio J., and Myrna L Sastré-Gutiérrez. “Interregional Inequality Dynamics in Mexico.” Spatial Economic Analysis 5, no. 3 (2010): 277–98.

**Comment by LSYS** [comment]
hi @sjsrey, thanks so much for the quick response. I've worked out basic technical details of each's spatial decomposition, so I'm aware they are computed differently and are not directly comparable. I just found it strange that using the same weights (the regime clusters) led to such distinct disagreement (in direction, nevermind magnitudes of the coefficients).

But thanks for sharing those two papers! Maybe they will shed some light for me. Closing this for now.

Thanks for developing so many OSS packages :)
